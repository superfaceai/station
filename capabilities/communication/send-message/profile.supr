"""
Send Message
Send message via IM such as Messanger, Slack or MS Teams.
"""

name = "communication/send-message"
version = "2.0.1"

"""
Send Message
Sends message to one destination
"""
usecase SendMessage {
  input {
    """
    Destination
    Representation of a messaging destination. Can be ID or channel name, depending on the provider capability.
    """
    destination! string!

    """
    Text
    Text of the message
    """
    text! string!
  }

  result {
    """
    Destination
    Identifier of the destination where the message was sent.
    """
    destination

    """
    Message ID
    Identifier of the message that was sent
    """
    messageId
  }

  error {
    """
    Title
    A short, human-readable summary of the problem type.
    """
    title!

    """
    A human-readable explanation specific to this occurrence of the problem.
    """
    detail
  }

  example Successful {
    input {
      destination = 'C1H9RESGL',
      text = 'Our app has 1,000,000 subscribers!',
    }

    result {
      destination = 'C1H9RESGL',
      messageId = "1503435956.000247",
    }
  }

  example Failed {
    input {
      destination = 'random',
      text = 'Message to nonexistent destination',
    }

    error {
      title = 'channel_not_found',
    }
  }
}
