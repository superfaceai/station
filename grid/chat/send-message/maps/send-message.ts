/* eslint-disable jest/no-export */

import { RecordingProcessOptions, SuperfaceTest } from '@superfaceai/testing';

export const sendMessageTest = (
  provider: string,
  destination: {
    valid: string;
    invalid: string;
  },
  options?: RecordingProcessOptions
): void => {
  describe(`chat/send-message/${provider}`, () => {
    let superface: SuperfaceTest;

    describe('SendMessage', () => {
      beforeAll(() => {
        superface = new SuperfaceTest({
          profile: 'chat/send-message',
          provider,
          useCase: 'SendMessage',
          testInstance: expect,
        });
      });

      describe('when specified destination does exist', () => {
        it('performs correctly', async () => {
          await expect(
            superface.run(
              {
                input: {
                  destination: destination.valid,
                  text: 'test',
                },
              },
              options
            )
          ).resolves.toMatchSnapshot();
        });
      });

      describe('when specified destination does not exist', () => {
        it('returns error', async () => {
          await expect(
            superface.run(
              {
                input: {
                  destination: destination.invalid,
                  text: 'test',
                },
              },
              options
            )
          ).resolves.toMatchSnapshot();
        });
      });
    });
  });
};
