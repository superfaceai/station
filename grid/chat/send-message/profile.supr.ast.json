{
  "kind": "ProfileDocument",
  "header": {
    "kind": "ProfileHeader",
    "scope": "chat",
    "name": "send-message",
    "version": {
      "major": 1,
      "minor": 0,
      "patch": 0
    },
    "location": {
      "start": {
        "line": 5,
        "column": 1,
        "charIndex": 66
      },
      "end": {
        "line": 6,
        "column": 18,
        "charIndex": 110
      }
    },
    "documentation": {
      "title": "Send a message",
      "description": "Send a text message to channel destination",
      "location": {
        "start": {
          "line": 1,
          "column": 1,
          "charIndex": 0
        },
        "end": {
          "line": 4,
          "column": 4,
          "charIndex": 65
        }
      }
    }
  },
  "definitions": [
    {
      "kind": "UseCaseDefinition",
      "useCaseName": "SendMessage",
      "input": {
        "kind": "UseCaseSlotDefinition",
        "value": {
          "kind": "ObjectDefinition",
          "fields": [
            {
              "kind": "FieldDefinition",
              "fieldName": "destination",
              "required": true,
              "type": {
                "kind": "NonNullDefinition",
                "type": {
                  "kind": "PrimitiveTypeName",
                  "name": "string",
                  "location": {
                    "start": {
                      "line": 18,
                      "column": 18,
                      "charIndex": 282
                    },
                    "end": {
                      "line": 18,
                      "column": 24,
                      "charIndex": 288
                    }
                  }
                },
                "location": {
                  "start": {
                    "line": 18,
                    "column": 18,
                    "charIndex": 282
                  },
                  "end": {
                    "line": 18,
                    "column": 25,
                    "charIndex": 289
                  }
                }
              },
              "location": {
                "start": {
                  "line": 18,
                  "column": 5,
                  "charIndex": 269
                },
                "end": {
                  "line": 18,
                  "column": 25,
                  "charIndex": 289
                }
              },
              "documentation": {
                "title": "Destination",
                "description": "Channel to send the message to",
                "location": {
                  "start": {
                    "line": 14,
                    "column": 5,
                    "charIndex": 202
                  },
                  "end": {
                    "line": 17,
                    "column": 8,
                    "charIndex": 264
                  }
                }
              }
            },
            {
              "kind": "FieldDefinition",
              "fieldName": "text",
              "required": true,
              "type": {
                "kind": "NonNullDefinition",
                "type": {
                  "kind": "PrimitiveTypeName",
                  "name": "string",
                  "location": {
                    "start": {
                      "line": 24,
                      "column": 11,
                      "charIndex": 358
                    },
                    "end": {
                      "line": 24,
                      "column": 17,
                      "charIndex": 364
                    }
                  }
                },
                "location": {
                  "start": {
                    "line": 24,
                    "column": 11,
                    "charIndex": 358
                  },
                  "end": {
                    "line": 24,
                    "column": 18,
                    "charIndex": 365
                  }
                }
              },
              "location": {
                "start": {
                  "line": 24,
                  "column": 5,
                  "charIndex": 352
                },
                "end": {
                  "line": 24,
                  "column": 18,
                  "charIndex": 365
                }
              },
              "documentation": {
                "title": "Text",
                "description": "Contents of the message",
                "location": {
                  "start": {
                    "line": 20,
                    "column": 5,
                    "charIndex": 299
                  },
                  "end": {
                    "line": 23,
                    "column": 8,
                    "charIndex": 347
                  }
                }
              }
            }
          ],
          "location": {
            "start": {
              "line": 13,
              "column": 9,
              "charIndex": 196
            },
            "end": {
              "line": 25,
              "column": 4,
              "charIndex": 369
            }
          }
        },
        "location": {
          "start": {
            "line": 13,
            "column": 3,
            "charIndex": 190
          },
          "end": {
            "line": 25,
            "column": 4,
            "charIndex": 369
          }
        }
      },
      "result": {
        "kind": "UseCaseSlotDefinition",
        "value": {
          "kind": "ObjectDefinition",
          "fields": [
            {
              "kind": "FieldDefinition",
              "fieldName": "messageId",
              "required": true,
              "type": {
                "kind": "PrimitiveTypeName",
                "name": "string",
                "location": {
                  "start": {
                    "line": 32,
                    "column": 16,
                    "charIndex": 472
                  },
                  "end": {
                    "line": 32,
                    "column": 22,
                    "charIndex": 478
                  }
                }
              },
              "location": {
                "start": {
                  "line": 32,
                  "column": 5,
                  "charIndex": 461
                },
                "end": {
                  "line": 32,
                  "column": 22,
                  "charIndex": 478
                }
              },
              "documentation": {
                "title": "Message ID",
                "description": "Identifier of the message that was sent",
                "location": {
                  "start": {
                    "line": 28,
                    "column": 5,
                    "charIndex": 386
                  },
                  "end": {
                    "line": 31,
                    "column": 8,
                    "charIndex": 456
                  }
                }
              }
            },
            {
              "kind": "FieldDefinition",
              "fieldName": "destination",
              "required": false,
              "type": {
                "kind": "PrimitiveTypeName",
                "name": "string",
                "location": {
                  "start": {
                    "line": 38,
                    "column": 17,
                    "charIndex": 590
                  },
                  "end": {
                    "line": 38,
                    "column": 23,
                    "charIndex": 596
                  }
                }
              },
              "location": {
                "start": {
                  "line": 38,
                  "column": 5,
                  "charIndex": 578
                },
                "end": {
                  "line": 38,
                  "column": 23,
                  "charIndex": 596
                }
              },
              "documentation": {
                "title": "Destination",
                "description": "Identifier of the destination where the message was sent.",
                "location": {
                  "start": {
                    "line": 34,
                    "column": 5,
                    "charIndex": 484
                  },
                  "end": {
                    "line": 37,
                    "column": 8,
                    "charIndex": 573
                  }
                }
              }
            },
            {
              "kind": "FieldDefinition",
              "fieldName": "rateLimit",
              "required": false,
              "type": {
                "kind": "ModelTypeName",
                "name": "RateLimit",
                "location": {
                  "start": {
                    "line": 44,
                    "column": 15,
                    "charIndex": 667
                  },
                  "end": {
                    "line": 44,
                    "column": 24,
                    "charIndex": 676
                  }
                }
              },
              "location": {
                "start": {
                  "line": 44,
                  "column": 5,
                  "charIndex": 657
                },
                "end": {
                  "line": 44,
                  "column": 24,
                  "charIndex": 676
                }
              },
              "documentation": {
                "title": "Rate Limit",
                "description": "Rate limit details.",
                "location": {
                  "start": {
                    "line": 40,
                    "column": 5,
                    "charIndex": 602
                  },
                  "end": {
                    "line": 43,
                    "column": 8,
                    "charIndex": 652
                  }
                }
              }
            }
          ],
          "location": {
            "start": {
              "line": 27,
              "column": 10,
              "charIndex": 380
            },
            "end": {
              "line": 45,
              "column": 4,
              "charIndex": 680
            }
          }
        },
        "location": {
          "start": {
            "line": 27,
            "column": 3,
            "charIndex": 373
          },
          "end": {
            "line": 45,
            "column": 4,
            "charIndex": 680
          }
        }
      },
      "error": {
        "kind": "UseCaseSlotDefinition",
        "value": {
          "kind": "ModelTypeName",
          "name": "Error",
          "location": {
            "start": {
              "line": 47,
              "column": 9,
              "charIndex": 690
            },
            "end": {
              "line": 47,
              "column": 14,
              "charIndex": 695
            }
          }
        },
        "location": {
          "start": {
            "line": 47,
            "column": 3,
            "charIndex": 684
          },
          "end": {
            "line": 47,
            "column": 14,
            "charIndex": 695
          }
        }
      },
      "examples": [
        {
          "kind": "UseCaseSlotDefinition",
          "value": {
            "kind": "UseCaseExample",
            "exampleName": "Successful",
            "input": {
              "kind": "UseCaseSlotDefinition",
              "value": {
                "kind": "ComlinkObjectLiteral",
                "fields": [
                  {
                    "kind": "ComlinkAssignment",
                    "key": [
                      "destination"
                    ],
                    "value": {
                      "kind": "ComlinkPrimitiveLiteral",
                      "value": "C1H9RESGL",
                      "location": {
                        "start": {
                          "line": 51,
                          "column": 21,
                          "charIndex": 752
                        },
                        "end": {
                          "line": 51,
                          "column": 32,
                          "charIndex": 763
                        }
                      }
                    },
                    "location": {
                      "start": {
                        "line": 51,
                        "column": 7,
                        "charIndex": 738
                      },
                      "end": {
                        "line": 51,
                        "column": 32,
                        "charIndex": 763
                      }
                    }
                  },
                  {
                    "kind": "ComlinkAssignment",
                    "key": [
                      "text"
                    ],
                    "value": {
                      "kind": "ComlinkPrimitiveLiteral",
                      "value": "Our app has 1,000,000 subscribers!",
                      "location": {
                        "start": {
                          "line": 52,
                          "column": 14,
                          "charIndex": 778
                        },
                        "end": {
                          "line": 52,
                          "column": 50,
                          "charIndex": 814
                        }
                      }
                    },
                    "location": {
                      "start": {
                        "line": 52,
                        "column": 7,
                        "charIndex": 771
                      },
                      "end": {
                        "line": 52,
                        "column": 50,
                        "charIndex": 814
                      }
                    }
                  }
                ],
                "location": {
                  "start": {
                    "line": 50,
                    "column": 11,
                    "charIndex": 730
                  },
                  "end": {
                    "line": 53,
                    "column": 6,
                    "charIndex": 821
                  }
                }
              },
              "location": {
                "start": {
                  "line": 50,
                  "column": 5,
                  "charIndex": 724
                },
                "end": {
                  "line": 53,
                  "column": 6,
                  "charIndex": 821
                }
              }
            },
            "result": {
              "kind": "UseCaseSlotDefinition",
              "value": {
                "kind": "ComlinkObjectLiteral",
                "fields": [
                  {
                    "kind": "ComlinkAssignment",
                    "key": [
                      "destination"
                    ],
                    "value": {
                      "kind": "ComlinkPrimitiveLiteral",
                      "value": "C1H9RESGL",
                      "location": {
                        "start": {
                          "line": 56,
                          "column": 21,
                          "charIndex": 856
                        },
                        "end": {
                          "line": 56,
                          "column": 32,
                          "charIndex": 867
                        }
                      }
                    },
                    "location": {
                      "start": {
                        "line": 56,
                        "column": 7,
                        "charIndex": 842
                      },
                      "end": {
                        "line": 56,
                        "column": 32,
                        "charIndex": 867
                      }
                    }
                  },
                  {
                    "kind": "ComlinkAssignment",
                    "key": [
                      "messageId"
                    ],
                    "value": {
                      "kind": "ComlinkPrimitiveLiteral",
                      "value": "1503435956.000247",
                      "location": {
                        "start": {
                          "line": 57,
                          "column": 19,
                          "charIndex": 887
                        },
                        "end": {
                          "line": 57,
                          "column": 38,
                          "charIndex": 906
                        }
                      }
                    },
                    "location": {
                      "start": {
                        "line": 57,
                        "column": 7,
                        "charIndex": 875
                      },
                      "end": {
                        "line": 57,
                        "column": 38,
                        "charIndex": 906
                      }
                    }
                  }
                ],
                "location": {
                  "start": {
                    "line": 55,
                    "column": 12,
                    "charIndex": 834
                  },
                  "end": {
                    "line": 58,
                    "column": 6,
                    "charIndex": 913
                  }
                }
              },
              "location": {
                "start": {
                  "line": 55,
                  "column": 5,
                  "charIndex": 827
                },
                "end": {
                  "line": 58,
                  "column": 6,
                  "charIndex": 913
                }
              }
            },
            "location": {
              "start": {
                "line": 49,
                "column": 11,
                "charIndex": 707
              },
              "end": {
                "line": 59,
                "column": 4,
                "charIndex": 917
              }
            }
          },
          "location": {
            "start": {
              "line": 49,
              "column": 3,
              "charIndex": 699
            },
            "end": {
              "line": 59,
              "column": 4,
              "charIndex": 917
            }
          }
        },
        {
          "kind": "UseCaseSlotDefinition",
          "value": {
            "kind": "UseCaseExample",
            "exampleName": "Failed",
            "input": {
              "kind": "UseCaseSlotDefinition",
              "value": {
                "kind": "ComlinkObjectLiteral",
                "fields": [
                  {
                    "kind": "ComlinkAssignment",
                    "key": [
                      "destination"
                    ],
                    "value": {
                      "kind": "ComlinkPrimitiveLiteral",
                      "value": "random",
                      "location": {
                        "start": {
                          "line": 63,
                          "column": 21,
                          "charIndex": 970
                        },
                        "end": {
                          "line": 63,
                          "column": 29,
                          "charIndex": 978
                        }
                      }
                    },
                    "location": {
                      "start": {
                        "line": 63,
                        "column": 7,
                        "charIndex": 956
                      },
                      "end": {
                        "line": 63,
                        "column": 29,
                        "charIndex": 978
                      }
                    }
                  },
                  {
                    "kind": "ComlinkAssignment",
                    "key": [
                      "text"
                    ],
                    "value": {
                      "kind": "ComlinkPrimitiveLiteral",
                      "value": "Message to nonexistent destination",
                      "location": {
                        "start": {
                          "line": 64,
                          "column": 14,
                          "charIndex": 993
                        },
                        "end": {
                          "line": 64,
                          "column": 50,
                          "charIndex": 1029
                        }
                      }
                    },
                    "location": {
                      "start": {
                        "line": 64,
                        "column": 7,
                        "charIndex": 986
                      },
                      "end": {
                        "line": 64,
                        "column": 50,
                        "charIndex": 1029
                      }
                    }
                  }
                ],
                "location": {
                  "start": {
                    "line": 62,
                    "column": 11,
                    "charIndex": 948
                  },
                  "end": {
                    "line": 65,
                    "column": 6,
                    "charIndex": 1036
                  }
                }
              },
              "location": {
                "start": {
                  "line": 62,
                  "column": 5,
                  "charIndex": 942
                },
                "end": {
                  "line": 65,
                  "column": 6,
                  "charIndex": 1036
                }
              }
            },
            "error": {
              "kind": "UseCaseSlotDefinition",
              "value": {
                "kind": "ComlinkObjectLiteral",
                "fields": [
                  {
                    "kind": "ComlinkAssignment",
                    "key": [
                      "title"
                    ],
                    "value": {
                      "kind": "ComlinkPrimitiveLiteral",
                      "value": "channel_not_found",
                      "location": {
                        "start": {
                          "line": 68,
                          "column": 15,
                          "charIndex": 1064
                        },
                        "end": {
                          "line": 68,
                          "column": 34,
                          "charIndex": 1083
                        }
                      }
                    },
                    "location": {
                      "start": {
                        "line": 68,
                        "column": 7,
                        "charIndex": 1056
                      },
                      "end": {
                        "line": 68,
                        "column": 34,
                        "charIndex": 1083
                      }
                    }
                  }
                ],
                "location": {
                  "start": {
                    "line": 67,
                    "column": 11,
                    "charIndex": 1048
                  },
                  "end": {
                    "line": 69,
                    "column": 6,
                    "charIndex": 1090
                  }
                }
              },
              "location": {
                "start": {
                  "line": 67,
                  "column": 5,
                  "charIndex": 1042
                },
                "end": {
                  "line": 69,
                  "column": 6,
                  "charIndex": 1090
                }
              }
            },
            "location": {
              "start": {
                "line": 61,
                "column": 11,
                "charIndex": 929
              },
              "end": {
                "line": 70,
                "column": 4,
                "charIndex": 1094
              }
            }
          },
          "location": {
            "start": {
              "line": 61,
              "column": 3,
              "charIndex": 921
            },
            "end": {
              "line": 70,
              "column": 4,
              "charIndex": 1094
            }
          }
        }
      ],
      "location": {
        "start": {
          "line": 12,
          "column": 1,
          "charIndex": 166
        },
        "end": {
          "line": 71,
          "column": 2,
          "charIndex": 1096
        }
      },
      "documentation": {
        "title": "Send Message",
        "description": "Sends message to one destination",
        "location": {
          "start": {
            "line": 8,
            "column": 1,
            "charIndex": 112
          },
          "end": {
            "line": 11,
            "column": 4,
            "charIndex": 165
          }
        }
      }
    },
    {
      "kind": "NamedModelDefinition",
      "modelName": "Error",
      "type": {
        "kind": "ObjectDefinition",
        "fields": [
          {
            "kind": "FieldDefinition",
            "fieldName": "title",
            "required": true,
            "location": {
              "start": {
                "line": 78,
                "column": 3,
                "charIndex": 1189
              },
              "end": {
                "line": 78,
                "column": 9,
                "charIndex": 1195
              }
            },
            "documentation": {
              "title": "Title",
              "description": "A short, human-readable summary of the problem type.",
              "location": {
                "start": {
                  "line": 74,
                  "column": 3,
                  "charIndex": 1114
                },
                "end": {
                  "line": 77,
                  "column": 6,
                  "charIndex": 1186
                }
              }
            }
          },
          {
            "kind": "FieldDefinition",
            "fieldName": "detail",
            "required": false,
            "location": {
              "start": {
                "line": 84,
                "column": 3,
                "charIndex": 1295
              },
              "end": {
                "line": 84,
                "column": 9,
                "charIndex": 1301
              }
            },
            "documentation": {
              "title": "Detail",
              "description": "A human-readable explanation specific to this occurrence of the problem.",
              "location": {
                "start": {
                  "line": 80,
                  "column": 3,
                  "charIndex": 1199
                },
                "end": {
                  "line": 83,
                  "column": 6,
                  "charIndex": 1292
                }
              }
            }
          },
          {
            "kind": "FieldDefinition",
            "fieldName": "rateLimit",
            "required": false,
            "type": {
              "kind": "ModelTypeName",
              "name": "RateLimit",
              "location": {
                "start": {
                  "line": 90,
                  "column": 13,
                  "charIndex": 1362
                },
                "end": {
                  "line": 90,
                  "column": 22,
                  "charIndex": 1371
                }
              }
            },
            "location": {
              "start": {
                "line": 90,
                "column": 3,
                "charIndex": 1352
              },
              "end": {
                "line": 90,
                "column": 22,
                "charIndex": 1371
              }
            },
            "documentation": {
              "title": "Rate Limit",
              "description": "Rate limit details.",
              "location": {
                "start": {
                  "line": 86,
                  "column": 3,
                  "charIndex": 1305
                },
                "end": {
                  "line": 89,
                  "column": 6,
                  "charIndex": 1349
                }
              }
            }
          }
        ],
        "location": {
          "start": {
            "line": 73,
            "column": 13,
            "charIndex": 1110
          },
          "end": {
            "line": 91,
            "column": 2,
            "charIndex": 1373
          }
        }
      },
      "location": {
        "start": {
          "line": 73,
          "column": 1,
          "charIndex": 1098
        },
        "end": {
          "line": 91,
          "column": 2,
          "charIndex": 1373
        }
      }
    },
    {
      "kind": "NamedModelDefinition",
      "modelName": "RateLimit",
      "type": {
        "kind": "ObjectDefinition",
        "fields": [
          {
            "kind": "FieldDefinition",
            "fieldName": "bucket",
            "required": false,
            "type": {
              "kind": "PrimitiveTypeName",
              "name": "string",
              "location": {
                "start": {
                  "line": 99,
                  "column": 10,
                  "charIndex": 1544
                },
                "end": {
                  "line": 99,
                  "column": 16,
                  "charIndex": 1550
                }
              }
            },
            "location": {
              "start": {
                "line": 99,
                "column": 3,
                "charIndex": 1537
              },
              "end": {
                "line": 99,
                "column": 16,
                "charIndex": 1550
              }
            },
            "documentation": {
              "title": "Bucket",
              "description": "Different parts of API may have different rate limits.\nBucket identifies to which part of API the rate limits apply.",
              "location": {
                "start": {
                  "line": 94,
                  "column": 3,
                  "charIndex": 1395
                },
                "end": {
                  "line": 98,
                  "column": 6,
                  "charIndex": 1534
                }
              }
            }
          },
          {
            "kind": "FieldDefinition",
            "fieldName": "totalRequests",
            "required": false,
            "type": {
              "kind": "PrimitiveTypeName",
              "name": "number",
              "location": {
                "start": {
                  "line": 105,
                  "column": 17,
                  "charIndex": 1644
                },
                "end": {
                  "line": 105,
                  "column": 23,
                  "charIndex": 1650
                }
              }
            },
            "location": {
              "start": {
                "line": 105,
                "column": 3,
                "charIndex": 1630
              },
              "end": {
                "line": 105,
                "column": 23,
                "charIndex": 1650
              }
            },
            "documentation": {
              "title": "Total Requests",
              "description": "Total requests available in the time window.",
              "location": {
                "start": {
                  "line": 101,
                  "column": 3,
                  "charIndex": 1554
                },
                "end": {
                  "line": 104,
                  "column": 6,
                  "charIndex": 1627
                }
              }
            }
          },
          {
            "kind": "FieldDefinition",
            "fieldName": "remainingRequests",
            "required": false,
            "type": {
              "kind": "PrimitiveTypeName",
              "name": "number",
              "location": {
                "start": {
                  "line": 111,
                  "column": 21,
                  "charIndex": 1756
                },
                "end": {
                  "line": 111,
                  "column": 27,
                  "charIndex": 1762
                }
              }
            },
            "location": {
              "start": {
                "line": 111,
                "column": 3,
                "charIndex": 1738
              },
              "end": {
                "line": 111,
                "column": 27,
                "charIndex": 1762
              }
            },
            "documentation": {
              "title": "Remaining Requests",
              "description": "Remaining requests available in the time window.",
              "location": {
                "start": {
                  "line": 107,
                  "column": 3,
                  "charIndex": 1654
                },
                "end": {
                  "line": 110,
                  "column": 6,
                  "charIndex": 1735
                }
              }
            }
          },
          {
            "kind": "FieldDefinition",
            "fieldName": "remainingRequestsPercentage",
            "required": false,
            "type": {
              "kind": "PrimitiveTypeName",
              "name": "number",
              "location": {
                "start": {
                  "line": 117,
                  "column": 31,
                  "charIndex": 1912
                },
                "end": {
                  "line": 117,
                  "column": 37,
                  "charIndex": 1918
                }
              }
            },
            "location": {
              "start": {
                "line": 117,
                "column": 3,
                "charIndex": 1884
              },
              "end": {
                "line": 117,
                "column": 37,
                "charIndex": 1918
              }
            },
            "documentation": {
              "title": "Remaining Requests available in percents",
              "description": "Remaining requests available in the time window in percents.",
              "location": {
                "start": {
                  "line": 113,
                  "column": 3,
                  "charIndex": 1766
                },
                "end": {
                  "line": 116,
                  "column": 6,
                  "charIndex": 1881
                }
              }
            }
          },
          {
            "kind": "FieldDefinition",
            "fieldName": "resetTimestamp",
            "required": false,
            "type": {
              "kind": "PrimitiveTypeName",
              "name": "number",
              "location": {
                "start": {
                  "line": 123,
                  "column": 18,
                  "charIndex": 2042
                },
                "end": {
                  "line": 123,
                  "column": 24,
                  "charIndex": 2048
                }
              }
            },
            "location": {
              "start": {
                "line": 123,
                "column": 3,
                "charIndex": 2027
              },
              "end": {
                "line": 123,
                "column": 24,
                "charIndex": 2048
              }
            },
            "documentation": {
              "title": "Reset Timestamp",
              "description": "Timestamp when the rate limits will reset (in Unix Timestamp ms format).",
              "location": {
                "start": {
                  "line": 119,
                  "column": 3,
                  "charIndex": 1922
                },
                "end": {
                  "line": 122,
                  "column": 6,
                  "charIndex": 2024
                }
              }
            }
          },
          {
            "kind": "FieldDefinition",
            "fieldName": "resetAfter",
            "required": false,
            "type": {
              "kind": "PrimitiveTypeName",
              "name": "number",
              "location": {
                "start": {
                  "line": 129,
                  "column": 14,
                  "charIndex": 2135
                },
                "end": {
                  "line": 129,
                  "column": 20,
                  "charIndex": 2141
                }
              }
            },
            "location": {
              "start": {
                "line": 129,
                "column": 3,
                "charIndex": 2124
              },
              "end": {
                "line": 129,
                "column": 20,
                "charIndex": 2141
              }
            },
            "documentation": {
              "title": "Reset After",
              "description": "Number of seconds until reset of rate limit",
              "location": {
                "start": {
                  "line": 125,
                  "column": 3,
                  "charIndex": 2052
                },
                "end": {
                  "line": 128,
                  "column": 6,
                  "charIndex": 2121
                }
              }
            }
          },
          {
            "kind": "FieldDefinition",
            "fieldName": "retryAfter",
            "required": false,
            "type": {
              "kind": "PrimitiveTypeName",
              "name": "number",
              "location": {
                "start": {
                  "line": 135,
                  "column": 14,
                  "charIndex": 2235
                },
                "end": {
                  "line": 135,
                  "column": 20,
                  "charIndex": 2241
                }
              }
            },
            "location": {
              "start": {
                "line": 135,
                "column": 3,
                "charIndex": 2224
              },
              "end": {
                "line": 135,
                "column": 20,
                "charIndex": 2241
              }
            },
            "documentation": {
              "title": "Retry After",
              "description": "Number of seconds until user can retry the request",
              "location": {
                "start": {
                  "line": 131,
                  "column": 3,
                  "charIndex": 2145
                },
                "end": {
                  "line": 134,
                  "column": 6,
                  "charIndex": 2221
                }
              }
            }
          }
        ],
        "location": {
          "start": {
            "line": 93,
            "column": 17,
            "charIndex": 1391
          },
          "end": {
            "line": 136,
            "column": 2,
            "charIndex": 2243
          }
        }
      },
      "location": {
        "start": {
          "line": 93,
          "column": 1,
          "charIndex": 1375
        },
        "end": {
          "line": 136,
          "column": 2,
          "charIndex": 2243
        }
      }
    }
  ],
  "location": {
    "start": {
      "line": 5,
      "column": 1,
      "charIndex": 66
    },
    "end": {
      "line": 136,
      "column": 2,
      "charIndex": 2243
    }
  },
  "astMetadata": {
    "astVersion": {
      "major": 1,
      "minor": 2,
      "patch": 0
    },
    "parserVersion": {
      "major": 1,
      "minor": 2,
      "patch": 0
    },
    "sourceChecksum": "e4f27bcd7e05a9d372e51aaa9d020daed50aa364e410058fb81d81fbc314bfdf"
  }
}