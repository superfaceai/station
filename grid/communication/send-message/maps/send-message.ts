/* eslint-disable jest/no-export */

import { SuperfaceTest } from '@superfaceai/testing';

import { nockConfig } from '../../../test-config';

export const sendMessage = (
  provider: string,
  params: { destination: string }
): void => {
  let superface: SuperfaceTest;

  beforeEach(() => {
    superface = new SuperfaceTest(
      {
        profile: 'communication/send-message',
        provider,
      },
      nockConfig
    );
  });

  describe(`communication/send-message/${provider}`, () => {
    it('performs correctly', async () => {
      await expect(
        superface.run({
          useCase: 'SendMessage',
          input: {
            destination: params.destination,
            text: 'Station communication/send-message test',
          },
        })
      ).resolves.toMatchSnapshot();
    });
  });
};
