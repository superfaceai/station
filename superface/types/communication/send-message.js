"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.communicationSendMessage = void 0;
var one_sdk_1 = require("@superfaceai/one-sdk");
var profile = {
    /**
     * Send Message
     * Sends message via IM such as Messanger, Slack or MS Teams...
     **/
    "SendMessage": one_sdk_1.typeHelper()
};
exports.communicationSendMessage = {
    "communication/send-message": profile
};
