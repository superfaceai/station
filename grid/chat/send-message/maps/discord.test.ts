import { sendMessageTest } from './send-message';

sendMessageTest('discord', {
  valid: '935962220104396885',
  invalid: '000000000000000000',
});
