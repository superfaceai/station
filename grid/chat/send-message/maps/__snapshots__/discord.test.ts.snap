// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chat/send-message/discord SendMessage when specified destination does exist performs correctly 1`] = `
Ok {
  "value": Object {
    "destination": "935962220104396885",
    "messageId": "941607966518411304",
    "rateLimit": Object {
      "bucket": "a06de0de4a08126315431cc0c55ee3dc",
      "remainingRequests": 4,
      "remainingRequestsPercentage": 80,
      "resetAfter": 5,
      "resetTimestamp": 1644567239000,
      "totalRequests": 5,
    },
  },
}
`;

exports[`chat/send-message/discord SendMessage when specified destination does not exist returns error 1`] = `
Err {
  "error": "HTTPError: Expected HTTP error
Properties: {
  \\"title\\": \\"Not found\\",
  \\"detail\\": \\"Unknown Channel\\",
  \\"rateLimit\\": {
    \\"bucket\\": \\"a06de0de4a08126315431cc0c55ee3dc\\",
    \\"totalRequests\\": 5,
    \\"remainingRequests\\": 4,
    \\"resetAfter\\": 5,
    \\"resetTimestamp\\": 1644567239000,
    \\"remainingRequestsPercentage\\": 80
  }
}
AST Path: definitions[0].statements[0].responseHandlers[1].statements[1]
Original Map Location: Line 35, column 7",
}
`;
